backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -50,4 +54,27 @@ public UserModel getUserById(
 		return userModelAssembler.toModel(user);
 	}
 
+
+    @PostMapping({ "/{userId}/email-validations"})
+    @Operation(summary = "Verify the status of a confirmation code.", operationId="verify-confirmation-code")
+    public ResponseEntity<?>  verifyConfirmationCodeStatus(
+            @NotBlank(message = "userId must not be null or blank")
+            @Parameter(description = "The ID of the user.", required = true)
+            @PathVariable String userId,
+            @NotBlank(message = "code must not be null or blank")
+            @Parameter(description = "The confirmation code.", required = true)
+            @RequestParam String code
+        ){
+			final var user = userService.getUserById(userId)
+				.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(userId)));
+			Assert.hasText(code, "id is required; it must not be null or blank");
```

Comment:
1. Assertions are usually done up front before any other code is executed.
2. There's a copy/paste error in the code.
3. The code should be sent in a request body.
4. The code should be validated using the JSR validation framework.

Reply:
Method rewritten in commit 2cebd2d6f0685660fa512c66c4a5da16869070e1..

Reply:
We should return a 404 if no user with the specified `userId` is found. The current implementation would return a 500.

Reply:
As of commit 70d88ff21299e8b6f77785dd0b22d60b10f1a733, a `ResourceNotFoundException`  is returned (with accompanying 404 `HttpStatus`) if the user is not found.

Reply:
Although `ResourceNotFoundException` is thrown, it is currently thrown at the service level. 

The exception should be thrown at the web/controller level instead as it is the controller's responsibility to handle web requests/responses (ie. returning 404 to the consumer). Take a look at `UsersController` and/or `SubscriptionsController` to see how that exception is currently being thrown.